diff -Naur tint2-0.11.orig//CMakeLists.txt tint2-0.11//CMakeLists.txt
--- tint2-0.11.orig//CMakeLists.txt	2010-06-15 23:38:14.000000000 +0200
+++ tint2-0.11//CMakeLists.txt	2011-08-02 22:50:27.397552832 +0200
@@ -26,6 +26,7 @@
 include_directories( ${PROJECT_BINARY_DIR}
                      src
                      src/battery
+                     src/bitcoin
                      src/clock
                      src/systray
                      src/taskbar
@@ -43,6 +44,7 @@
              src/panel.c
              src/server.c
              src/tint.c
+             src/bitcoin/bitcoin.c
              src/clock/clock.c
              src/systray/systraybar.c
              src/taskbar/task.c
diff -Naur tint2-0.11.orig//src/bitcoin/bitcoin.c tint2-0.11//src/bitcoin/bitcoin.c
--- tint2-0.11.orig//src/bitcoin/bitcoin.c	1970-01-01 01:00:00.000000000 +0100
+++ tint2-0.11//src/bitcoin/bitcoin.c	2011-08-03 11:17:35.940055319 +0200
@@ -0,0 +1,214 @@
+/**************************************************************************
+*
+* Tint2 : bitcoin
+*
+* Copyright (C) 2011 xHire (xhire@mujmalysvet.cz)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License version 2
+* as published by the Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+**************************************************************************/
+
+#include <string.h>
+#include <stdio.h>
+#include <cairo.h>
+#include <cairo-xlib.h>
+#include <pango/pangocairo.h>
+#include <stdlib.h>
+
+#include "window.h"
+#include "server.h"
+#include "area.h"
+#include "panel.h"
+#include "taskbar.h"
+#include "bitcoin.h"
+#include "timer.h"
+#include "common.h"
+
+
+struct timeval time_bitcoin;
+PangoFontDescription *bitcoin_font_desc;
+static char buf[40];
+int bitcoin_enabled;
+static timeout* bitcoin_timeout;
+
+
+void default_bitcoin()
+{
+	bitcoin_enabled = 0;
+	bitcoin_timeout = 0;
+	bitcoin_font_desc = 0;
+}
+
+void cleanup_bitcoin()
+{
+	if (bitcoin_font_desc) {
+		pango_font_description_free(bitcoin_font_desc);
+	}
+}
+
+
+void update_bitcoins(void* arg)
+{
+	FILE *f;
+	// remember old_sec because after suspend/hibernate the bitcoin should be updated directly, and not
+	// on next minute change
+	time_t old_sec = time_bitcoin.tv_sec;
+	gettimeofday(&time_bitcoin, 0);
+	if (time_bitcoin.tv_sec % 60 == 0 || time_bitcoin.tv_sec - old_sec > 60) {
+		int i;
+		for (i = 0; i < nb_panel; i++) {
+			panel1[i].bitcoin.area.resize = 1;
+		}
+		panel_refresh = 1;
+
+		/* update bitcoin rate */
+		system("/usr/bin/tint2-bitcoinrate");
+		f = fopen("/tmp/btcrate", "r");
+		if (f) {
+			int res;
+			res = fread(&buf, sizeof(char), 37, f);
+			buf[res] = ' ';
+			buf[res + 1] = '$';
+			buf[res + 2] = '\0';
+			fclose(f);
+		}
+	}
+}
+
+
+void init_bitcoin()
+{
+	if (bitcoin_timeout == 0) {
+		bitcoin_timeout = add_timeout(10, 60 * 1000, update_bitcoins, 0);
+	}
+}
+
+
+void init_bitcoin_panel(void *p)
+{
+	Panel *panel = (Panel*) p;
+	Bitcoin *bitcoin = &panel->bitcoin;
+	int bitcoin_height, bitcoin_height_ink;
+
+	bitcoin->area.parent = p;
+	bitcoin->area.panel = p;
+	bitcoin->area._draw_foreground = draw_bitcoin;
+	bitcoin->area._resize = resize_bitcoin;
+	bitcoin->area.resize = 1;
+	bitcoin->area.redraw = 1;
+	bitcoin->area.on_screen = 1;
+
+	get_text_size(bitcoin_font_desc, &bitcoin_height_ink, &bitcoin_height, panel->area.height, buf, strlen(buf));
+
+	if (panel_horizontal) {
+		// panel horizonal => fixed height and posy
+		bitcoin->area.posy = panel->area.bg->border.width + panel->area.paddingy;
+		bitcoin->area.height = panel->area.height - (2 * bitcoin->area.posy);
+	} else {
+		// panel vertical => fixed width, height, posy and posx
+		bitcoin->area.posy = panel->area.bg->border.width + panel->area.paddingxlr;
+		bitcoin->area.height = (2 * bitcoin->area.paddingxlr) + bitcoin_height;
+		bitcoin->area.posx = panel->area.bg->border.width + panel->area.paddingy;
+		bitcoin->area.width = panel->area.width - (2 * panel->area.bg->border.width) - (2 * panel->area.paddingy);
+	}
+
+	bitcoin->posy = (bitcoin->area.height - bitcoin_height) / 2;
+}
+
+
+void draw_bitcoin(void *obj, cairo_t *c)
+{
+	Bitcoin *bitcoin = obj;
+	PangoLayout *layout;
+
+	layout = pango_cairo_create_layout(c);
+
+	// draw layout
+	pango_layout_set_font_description(layout, bitcoin_font_desc);
+	pango_layout_set_width(layout, bitcoin->area.width * PANGO_SCALE);
+	pango_layout_set_alignment(layout, PANGO_ALIGN_CENTER);
+	pango_layout_set_text(layout, buf, strlen(buf));
+
+	cairo_set_source_rgba(c, bitcoin->font.color[0], bitcoin->font.color[1], bitcoin->font.color[2], bitcoin->font.alpha);
+
+	pango_cairo_update_layout(c, layout);
+	cairo_move_to(c, 0, bitcoin->posy);
+	pango_cairo_show_layout(c, layout);
+
+	g_object_unref (layout);
+}
+
+
+void resize_bitcoin(void *obj)
+{
+	Bitcoin *bitcoin = obj;
+	PangoLayout *layout;
+	int bitcoin_width;
+
+	bitcoin->area.redraw = 1;
+	bitcoin_width = 0;
+
+	// vertical panel doesn't adjust width
+	if (!panel_horizontal) {
+		return;
+	}
+
+	//printf("  resize_bitcoin\n");
+	cairo_surface_t *cs;
+	cairo_t *c;
+	Pixmap pmap;
+	pmap = XCreatePixmap(server.dsp, server.root_win, bitcoin->area.width, bitcoin->area.height, server.depth);
+
+	cs = cairo_xlib_surface_create(server.dsp, pmap, server.visual, bitcoin->area.width, bitcoin->area.height);
+	c = cairo_create(cs);
+	layout = pango_cairo_create_layout(c);
+
+	// check width
+	pango_layout_set_font_description(layout, bitcoin_font_desc);
+	pango_layout_set_indent(layout, 0);
+	pango_layout_set_text(layout, buf, strlen(buf));
+	pango_layout_get_pixel_size(layout, &bitcoin_width, NULL);
+
+	bitcoin_width += (2 * bitcoin->area.paddingxlr) + (2 * bitcoin->area.bg->border.width);
+
+	Panel *panel = ((Area *) obj)->panel;
+	if (bitcoin_width > bitcoin->area.width || bitcoin_width < (bitcoin->area.width - 6)) {
+		// resize bitcoin
+		// we try to limit the number of resize
+		// printf("bitcoin_widget_width %d, bitcoin_width %d\n", bitcoin->area.width, bitcoin_width);
+		bitcoin->area.width = bitcoin_width + 1;
+
+		// resize other objects on panel
+		panel->area.resize = 1;
+		panel->clock.area.resize = 1;
+#ifdef ENABLE_BATTERY
+		panel->battery.area.resize = 1;
+#endif
+		systray.area.resize = 1;
+		panel_refresh = 1;
+	}
+	bitcoin->area.posx = panel->area.width - bitcoin->area.width - panel->area.paddingxlr - panel->area.bg->border.width;
+	if (panel->clock.area.on_screen) {
+		bitcoin->area.posx -= (panel->clock.area.width + panel->area.paddingx);
+	}
+#ifdef ENABLE_BATTERY
+	if (panel->battery.area.on_screen) {
+		bitcoin->area.posx -= (panel->battery.area.width + panel->area.paddingx);
+	}
+#endif
+
+
+	g_object_unref(layout);
+	cairo_destroy(c);
+	cairo_surface_destroy(cs);
+	XFreePixmap(server.dsp, pmap);
+}
diff -Naur tint2-0.11.orig//src/bitcoin/bitcoin.h tint2-0.11//src/bitcoin/bitcoin.h
--- tint2-0.11.orig//src/bitcoin/bitcoin.h	1970-01-01 01:00:00.000000000 +0100
+++ tint2-0.11//src/bitcoin/bitcoin.h	2011-08-03 11:18:39.106251241 +0200
@@ -0,0 +1,43 @@
+/**************************************************************************
+* Copyright (C) 2011 xHire (xhire@mujmalysvet.cz)
+*
+* Each panel uses its own drawing data.
+*
+**************************************************************************/
+
+#ifndef BITCOIN_H
+#define BITCOIN_H
+
+#include <sys/time.h>
+#include "common.h"
+#include "area.h"
+
+
+typedef struct Bitcoin {
+	// always start with area
+	Area area;
+
+	Color font;
+	int posy;
+} Bitcoin;
+
+
+extern PangoFontDescription *bitcoin_font_desc;
+extern int bitcoin_enabled;
+
+
+// default global data
+void default_bitcoin();
+
+// freed memory
+void cleanup_bitcoin();
+
+// initialize bitcoin: y position, precision, ...
+void init_bitcoin();
+void init_bitcoin_panel(void *panel);
+
+void draw_bitcoin(void *obj, cairo_t *c);
+
+void resize_bitcoin(void *obj);
+
+#endif
diff -Naur tint2-0.11.orig//src/clock/clock.c tint2-0.11//src/clock/clock.c
--- tint2-0.11.orig//src/clock/clock.c	2010-06-11 20:47:26.000000000 +0200
+++ tint2-0.11//src/clock/clock.c	2011-08-02 23:24:04.612606165 +0200
@@ -274,6 +274,7 @@
 
 		// resize other objects on panel
 		panel->area.resize = 1;
+		panel->bitcoin.area.resize = 1;
 #ifdef ENABLE_BATTERY
 		panel->battery.area.resize = 1;
 #endif
diff -Naur tint2-0.11.orig//src/config.c tint2-0.11//src/config.c
--- tint2-0.11.orig//src/config.c	2010-06-12 14:25:21.000000000 +0200
+++ tint2-0.11//src/config.c	2011-08-02 23:59:53.226719595 +0200
@@ -41,6 +41,7 @@
 #include "task.h"
 #include "taskbar.h"
 #include "systraybar.h"
+#include "bitcoin.h"
 #include "clock.h"
 #include "config.h"
 #include "window.h"
@@ -289,6 +290,37 @@
 			panel_layer = NORMAL_LAYER;
 	}
 
+	/* Bitcoin */
+	else if (strcmp(key, "bitcoin") == 0) {
+		if (atoi(value) == 1)
+			bitcoin_enabled = 1;
+	}
+	else if (strcmp(key, "bitcoin_font") == 0) {
+		bitcoin_font_desc = pango_font_description_from_string (value);
+	}
+	else if (strcmp(key, "bitcoin_font_color") == 0) {
+		extract_values(value, &value1, &value2, &value3);
+		get_color(value1, panel_config.bitcoin.font.color);
+		if (value2) panel_config.bitcoin.font.alpha = (atoi(value2) / 100.0);
+		else panel_config.bitcoin.font.alpha = 0.5;
+	}
+	else if (strcmp(key, "bitcoin_padding") == 0) {
+		extract_values(value, &value1, &value2, &value3);
+		panel_config.bitcoin.area.paddingxlr = panel_config.bitcoin.area.paddingx = atoi(value1);
+		if (value2) panel_config.bitcoin.area.paddingy = atoi(value2);
+		if (value3) panel_config.bitcoin.area.paddingx = atoi(value3);
+	}
+	else if (strcmp(key, "bitcoin_background_id") == 0) {
+		int id = atoi(value);
+		id = (id < backgrounds->len && id >= 0) ? id : 0;
+		panel_config.bitcoin.area.bg = &g_array_index(backgrounds, Background, id);
+	}
+	else if (strcmp(key, "bitcoin_hide") == 0) {
+		percentage_hide = atoi(value);
+		if (percentage_hide == 0)
+			percentage_hide = 101;
+	}
+
 	/* Battery */
 	else if (strcmp (key, "battery") == 0) {
 #ifdef ENABLE_BATTERY
diff -Naur tint2-0.11.orig//src/panel.c tint2-0.11//src/panel.c
--- tint2-0.11.orig//src/panel.c	2010-06-26 14:35:35.000000000 +0200
+++ tint2-0.11//src/panel.c	2011-08-02 22:44:22.250261711 +0200
@@ -135,6 +135,7 @@
 	init_tooltip();
 	init_systray();
 	init_clock();
+	init_bitcoin();
 #ifdef ENABLE_BATTERY
 	init_battery();
 #endif
@@ -170,6 +171,10 @@
 			init_clock_panel(p);
 			p->area.list = g_slist_append(p->area.list, &p->clock);
 		}
+		if (bitcoin_enabled) {
+			init_bitcoin_panel(p);
+			p->area.list = g_slist_append(p->area.list, &p->bitcoin);
+		}
 #ifdef ENABLE_BATTERY
 		if (battery_enabled) {
 			init_battery_panel(p);
@@ -303,6 +308,8 @@
 		taskbar_width = panel->area.width - (2 * panel->area.paddingxlr) - (2 * panel->area.bg->border.width);
 		if (panel->clock.area.on_screen && panel->clock.area.width)
 			taskbar_width -= (panel->clock.area.width + panel->area.paddingx);
+		if (panel->bitcoin.area.on_screen && panel->bitcoin.area.width)
+			taskbar_width -= (panel->bitcoin.area.width + panel->area.paddingx);
 	#ifdef ENABLE_BATTERY
 		if (panel->battery.area.on_screen && panel->battery.area.width)
 			taskbar_width -= (panel->battery.area.width + panel->area.paddingx);
@@ -340,6 +347,8 @@
 		taskbar_height = panel->area.height - (2 * panel->area.paddingxlr) - (2 * panel->area.bg->border.width);
 		if (panel->clock.area.on_screen && panel->clock.area.height)
 			taskbar_height -= (panel->clock.area.height + panel->area.paddingx);
+		if (panel->bitcoin.area.on_screen && panel->bitcoin.area.height)
+			taskbar_height -= (panel->bitcoin.area.height + panel->area.paddingx);
 	#ifdef ENABLE_BATTERY
 		if (panel->battery.area.on_screen && panel->battery.area.height)
 			taskbar_height -= (panel->battery.area.height + panel->area.paddingx);
diff -Naur tint2-0.11.orig//src/panel.h tint2-0.11//src/panel.h
--- tint2-0.11.orig//src/panel.h	2010-06-26 14:35:35.000000000 +0200
+++ tint2-0.11//src/panel.h	2011-08-02 22:44:57.303335003 +0200
@@ -16,6 +16,7 @@
 
 #include "common.h"
 #include "clock.h"
+#include "bitcoin.h"
 #include "task.h"
 #include "taskbar.h"
 #include "systraybar.h"
@@ -102,6 +103,10 @@
 	Clock clock;
 
 	// --------------------------------------------------
+	// bitcoin
+	Bitcoin bitcoin;
+
+	// --------------------------------------------------
 	// battery
 #ifdef ENABLE_BATTERY
 	Battery battery;
diff -Naur tint2-0.11.orig//src/systray/systraybar.c tint2-0.11//src/systray/systraybar.c
--- tint2-0.11.orig//src/systray/systraybar.c	2010-06-09 17:31:18.000000000 +0200
+++ tint2-0.11//src/systray/systraybar.c	2011-08-02 23:44:59.300004245 +0200
@@ -162,6 +162,8 @@
 		systray.area.posx = panel->area.width - panel->area.bg->border.width - panel->area.paddingxlr - systray.area.width;
 		if (panel->clock.area.on_screen)
 			systray.area.posx -= (panel->clock.area.width + panel->area.paddingx);
+		if (panel->bitcoin.area.on_screen)
+			systray.area.posx -= (panel->bitcoin.area.width + panel->area.paddingx);
 #ifdef ENABLE_BATTERY
 		if (panel->battery.area.on_screen)
 			systray.area.posx -= (panel->battery.area.width + panel->area.paddingx);
